tests/e2e_tests/test_strict_mode_memory.py

```
@@ -0,0 +1,230 @@
+def _try_upsert(client: ClientUtils, points: list[models.PointStruct]) -> Optional[str]:
+    """Upsert a batch; return the error string if rejected, else None."""
+    try:
+        client.client.upsert(
+            collection_name=COLLECTION_NAME,
+            points=points,
+            wait=True,
+        )
+        return None
+    except UnexpectedResponse as exc:
+        return str(exc)
+    except Exception as exc:  # qdrant-client may wrap differently depending on version
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)